token/jwt/token_test.go

```diff
@@ -0,0 +1,379 @@
+			tokenString: "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJmb28iOiJiYXIifQ.FhkiHkoESI_cG3NPigFrxEk9Z60_oXrOT2vGm9Pn6RDgYNovYORQmmA0zs1AoAOf09ly2Nx2YAg6ABqAYga1AcMFkJljwxTT5fYphTuqpWdy4BELeSYJx5Ty2gmr8e7RonuUztrdD5WfPqLKMm1Ozp_T6zALpRmwTIW0QPnaBXaQD90FplAg46Iy1UlDKr-Eupy0i5SLch5Q-p2ZpaL_5fnTIUDlxC3pWhJTyx_71qDI-mAA_5lE_VdroOeflG56sSmDxopPEG3bFlSu1eowyBfxtu0_CuVd-M42RU75Zc4Gsj6uV77MBtbMrf4_7M_NUTSgoIF3fRqxrj0NzihIBg",
+			keyfunc:     publicECDSAKey,
+			claims:      MapClaims{"foo": "bar"},
+			valid:       false,
+			errors:      ValidationErrorSignatureInvalid,
+		},
+		{
+			name:        "nokey, expected RSA got ECDSA",
+			tokenString: "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJmb28iOiJiYXIifQ.FhkiHkoESI_cG3NPigFrxEk9Z60_oXrOT2vGm9Pn6RDgYNovYORQmmA0zs1AoAOf09ly2Nx2YAg6ABqAYga1AcMFkJljwxTT5fYphTuqpWdy4BELeSYJx5Ty2gmr8e7RonuUztrdD5WfPqLKMm1Ozp_T6zALpRmwTIW0QPnaBXaQD90FplAg46Iy1UlDKr-Eupy0i5SLch5Q-p2ZpaL_5fnTIUDlxC3pWhJTyx_71qDI-mAA_5lE_VdroOeflG56sSmDxopPEG3bFlSu1eowyBfxtu0_CuVd-M42RU75Zc4Gsj6uV77MBtbMrf4_7M_NUTSgoIF3fRqxrj0NzihIBg",
+			keyfunc:     emptyKeyFunc,
+			claims:      MapClaims{"foo": "bar"},
+			valid:       false,
+			errors:      ValidationErrorSignatureInvalid,
+		},
+		{
+			name:        "key does not match",
+			tokenString: "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJmb28iOiJiYXIifQ.FhkiHkoESI_cG3NPigFrxEk9Z60_oXrOT2vGm9Pn6RDgYNovYORQmmA0zs1AoAOf09ly2Nx2YAg6ABqAYga1AcMFkJljwxTT5fYphTuqpWdy4BELeSYJx5Ty2gmr8e7RonuUztrdD5WfPqLKMm1Ozp_T6zALpRmwTIW0QPnaBXaQD90FplAg46Iy1UlDKr-Eupy0i5SLch5Q-p2ZpaL_5fnTIUDlxC3pWhJTyx_71qDI-mAA_5lE_VdroOeflG56sSmDxopPEG3bFlSu1eowyBfxtu0_CuVd-M42RU75Zc4Gsj6uV77MBtbMrf4_7M_NUTSgoIF3fRqxrj0NzihIBg",
+			keyfunc:     randomKey,
+			claims:      MapClaims{"foo": "bar"},
+			valid:       false,
+			errors:      ValidationErrorSignatureInvalid,
+		},
+		{
+			name:        "used before issued",
+			tokenString: "", // autogen
+			keyfunc:     defaultKeyFunc,
+			claims:      MapClaims{"foo": "bar", "iat": float64(time.Now().Unix() + 500)},
+			valid:       false,
+			errors:      ValidationErrorIssuedAt,
+		},
+		{
+			name:        "valid ECDSA signing method",
+			tokenString: "",
+			keyfunc:     publicECDSAKey,
+			claims:      MapClaims{"foo": "bar"},
+			valid:       true,
+			errors:      0,
+			signingKey:  defaultES256PrivateKey,
+			method:      jose.ES256,
+		},
+		{
+			name:        "valid NONE signing method",
+			tokenString: "",
+			keyfunc:     noneKey,
+			claims:      MapClaims{"foo": "bar"},
+			valid:       true,
+			errors:      0,
+			signingKey:  UnsafeAllowNoneSignatureType,
+			method:      SigningMethodNone,
+		},
+		{
+			name:        "should fail, valid NONE token with invalid verification key",
+			tokenString: "",
+			keyfunc:     defaultKeyFunc,
+			claims:      MapClaims{"foo": "bar"},
+			valid:       false,
+			errors:      ValidationErrorSignatureInvalid,
+			signingKey:  UnsafeAllowNoneSignatureType,
+			method:      SigningMethodNone,
+		},
```

Comment:
Could you please add one more test here:

1. `should fail, expected ECDSA but got NONE` which is the invert to https://github.com/ory/fosite/pull/593/files#diff-62425fcddec1b39f6b477e442647c937f444850b79d1acdbfc8b2ad7289d7a31R210

Reply:
done